Patch from https://github.com/shlomif/perl-XML-LibXSLT/commit/40fdc7f0.patch

From 40fdc7f0443949d9556f0ecf15eb7777d543cee1 Mon Sep 17 00:00:00 2001
From: Florian Weimer <fweimer@redhat.com>
Date: Thu, 9 Feb 2023 19:28:14 +0100
Subject: [PATCH] Avoid C implicit function declaration in Makefile.PL (C99
 compat)

Future compilers are likely not to support implicit function
declarations.  Add a fake prototype so that the probes will
not always fail with such compilers.
---
 Makefile.PL | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/Makefile.PL b/Makefile.PL
index 079f8ea..d720b48 100644
--- a/Makefile.PL
+++ b/Makefile.PL
@@ -463,10 +463,11 @@ SRC
             }
         }
         else {
-
+	    # Use a fake prototype in the style of autoconf.
             $result = try_link(<<"SRC", $libs);
-blank() { return 0; }
-int t() { ${func}(); return 0; }
+char blank(void) { return 0; }
+char ${func}(void);
+int t(void) { ${func}(); return 0; }
 SRC
         }
     }
